crates/net/network/src/network.rs

```diff
@@ -413,7 +418,12 @@ impl<N: NetworkPrimitives> NetworkSyncUpdater for NetworkHandle<N> {
 
     /// Update the status of the node.
     fn update_status(&self, head: Head) {
-        self.send_message(NetworkHandleMessage::StatusUpdate { head });
+        // Derive earliest_block as the Paris (Merge) activation block:
+        let merge_block = ChainSpec::default().paris_block().unwrap_or(0);
+        let earliest = if head.number >= merge_block { U64::from(merge_block) } else { U64::ZERO };
```

Comment:
I set earliest to the pre-merge block since that was called out in the eip https://eips.ethereum.org/EIPS/eip-7642#:~:text=In%20the%20history%20expiry%20working%20group%2C%20it%20was%20decided%20that%20clients%20may%20drop%20pre%2Dmerge%20history%20from%20their%20storage%20after%20May%201

lmk if another implementation is preferable